src/User.cpp

```diff
@@ -164,3 +184,214 @@ User User::getUserInformationFromDatabase(User &user) {
+
+    std::string stringPosts;
+    std::vector<std::shared_ptr<Post>> posts = profile.getPosts();
+    for (const auto &post: posts) {
+        if (stringPosts.empty()) {
+            stringPosts.append(std::to_string(post->getId()));
+        } else {
+            stringPosts = stringPosts
+                    .append(std::string(","))
+                    .append(std::to_string(post->getId()));
+        }
+    }
+
+    std::string stringGroups;
+    std::vector<std::shared_ptr<Group>> groups = profile.getGroups();
+    for (const auto &group: groups) {
+        if (stringGroups.empty()) {
+            stringGroups.append(group->getName());
+        } else {
+            stringGroups = stringGroups
+                    .append(std::string(","))
+                    .append(group->getName());
+        }
+    }
+
+    std::string stringFollowers;
+    std::vector<std::shared_ptr<User>> followers = profile.getFollowers();
+    for (const auto &follower: followers) {
+        if (stringFollowers.empty()) {
+            stringFollowers.append(follower->getUsername());
+        } else {
+            stringFollowers = stringFollowers
+                    .append(std::string(","))
+                    .append(follower->getUsername());
+        }
+    }
+
+    std::string stringStudies;
+    std::vector<std::string> studies = profile.getStudies();
+    for (const auto &study: studies) {
+        if (stringStudies.empty()) {
+            stringStudies = stringStudies.append(study);
+        } else {
+            stringStudies = stringStudies
+                    .append(std::string(","))
+                    .append(study);
+        }
+    }
+
+    std::string stringLivingPlaces;
+    std::vector<std::string> livingPlaces = profile.getLivingPlaces();
+    for (const auto &livingPlace: livingPlaces) {
+        if (stringLivingPlaces.empty()) {
+            stringLivingPlaces.append(livingPlace);
+        } else {
+            stringLivingPlaces = stringLivingPlaces
+                    .append(std::string(","))
+                    .append(livingPlace);
+        }
+    }
+
+    std::string data = id
+            .append("|").append(stringPosts)
+            .append("|").append(stringGroups)
+            .append("|").append(stringFollowers)
+            .append("|").append(stringStudies)
+            .append("|").append(stringLivingPlaces);
+
+    file.open("profiles.txt", std::ios::out | std::ios::app);
+
+    file << std::endl << data;
+
+    file.close();
+}
+
+
+// Implementarea pentru register
+void User::registerUser() {
+    std::string username;
+    std::cout << "Deci nu ai un cont. Te rugam sa introduci numele de utilizator dorit." << std::endl;
+    std::cin >> username;
+    User user;
+    user.setUsername(username);
+    bool availabilityUser = User::checkUserAvailability(user);
+    if (availabilityUser) {
+        std::cout << "Se pare ca ai cont, deci nu putem sa cream alt cont cu acelasi nume de utilizator." << std::endl;
```

Comment:
Nu e corect, poate fi vorba de alt user. Un mesaj mai corect e "Nume indisponibil". La login tot așa, trebuie doar să zici "Combinația user/parolă incorectă", atât. Nu zici dacă userul e corect și parola greșită pt că ăsta e bug de securitate și doar îi ajuți pe cei care vor să spargă conturi